core/notifications/src/app/mod.rs

```diff
@@ -25,6 +25,17 @@ impl NotificationsApp {
         }
     }
 
+    pub async fn notification_settings_for_account(
+        &self,
+        account_id: GaloyAccountId,
+    ) -> Result<AccountNotificationSettings, ApplicationError> {
+        if let Some(settings) = self.settings.find_for_account_id(&account_id).await? {
```

Comment:
probably because I dont like the else blocks :) but... can we replace this with `map_or_else` or something different? (applies to all the if/else blocks in this PR)

Reply:
I didn't introduce this pattern, and I'm not sure what is canonical for rust. @bodymindarts any thoughts?

Reply:
I agree with @dolcalmi 
```
        let mut user_settings = self
            .settings
            .find_for_user_id(&user_id)
            .await?
            .unwrap_or_else(|| UserNotificationSettings::new(user_id));
```

looks nicer.

Reply:
Made this change